### PR TITLE
Remove HIDDEN_LANGUAGES, sync with .po files and product-details.

### LINKED ISSUE #6534: Remove concept of HIDDEN_LANGUAGES
We don't have the l10n dashboard anymore so we should just make languages available whenever they are available via Pontoon / as .po files.

We have a -dev and -stage concept for people to review strings and deploying weekly gives translators enough time to improve translations incrementally.

Also, no idea who decides when a language get's promoted so let's save all that hassle and just promote all languages :)

### LINKED ISSUE #6535: Sync AMO_LANGUAGES with what's in locale/
There's a sync difference between `AMO_LANGUAGES` and what's in `locale/`. Let's sync these two and get translators to work on things which should be easier once they see stuff.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #6534, Fixes #6535

Actually there isn't much syncing possible since we're quite restricted with what product-details supports. So this is now the most up-to-date list of all languages from our .po files, hidden languages merged with what product-details supports.

I didn't see much reason in keeping HIDDEN_LANGUAGES lying around in the code when we don't have anything using it and delaying language promotion feels wrong to me.

Don't ask me what that `locale/testing.milos` file is, hasn't been touched in years, doesn't look like it's useful so I got rid of it :)

### CODE REVIEW

File: src/olympia/lib/settings_base.py [hunk 1]
```diff
@@ -153,21 +153,62 @@ def cors_endpoint_overrides(internal, public):
...
+    'sv-SE',  # Swedish
+    'uk',  # Ukrainian
+    'ur',  # Urdu
+    'vi',  # Vietnamese
+    'zh-CN',  # Chinese (Simplified)
+    'zh-TW',  # Chinese (Traditional)
 )
 
+# Bidirectional languages.
+# Locales in here *must* be in `AMO_LANGUAGES` too.
```
Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: is it worth an assert?  Otherwise a test

**Comment by EnTeQuAk** [comment]
>  What's with locale/testing.milos?

From my description:

> Don't ask me what that locale/testing.milos file is, hasn't been touched in years, doesn't look like it's useful so I got rid of it :)

File: src/olympia/lib/settings_base.py (hunk 1 above)
Comment (EnTeQuAk) [reply src/olympia/lib/settings_base.py]: Added in 25b842ab92d3a7dbae15cafff7fb68a5ca7dc95b and fixed in cf5e0a674e150ed74e23c91d89bb25e1ea80991b

**Comment by EnTeQuAk** [comment]
NOTE: with 2ddf5fe I removed our custom locale settings from our test settings which I talked talked about with @diox in the past already and now seemed a good time to do.

**Pull request merged**